providers/FoliaProvider/src/main/java/com/palmergames/bukkit/towny/scheduling/impl/FoliaTaskScheduler.java

```diff
@@ -114,4 +114,12 @@ public boolean isRegionThread(@NotNull Location location) {
 	public @NotNull ScheduledTask runAsyncRepeating(@NotNull Runnable runnable, long delay, long period) {
 		return new FoliaScheduledTask(this.asyncScheduler.runAtFixedRate(this.plugin, task -> runnable.run(), delay * 50L, period * 50L, TimeUnit.MILLISECONDS));
 	}
+
+	/**
+	 * Cancels all active tasks that have been scheduled by {@code this.plugin}
+	 */
+	public void cancelTasks() {
+		this.asyncScheduler.cancelTasks(this.plugin);
+		this.globalRegionScheduler.cancelTasks(this.plugin);
```

Comment:
Will this result in the `townyUniverse.finishTasks();` being cut off earlier than we want?

Reply:
We wait for the backupFuture to finish so at that time it's already finished, it also doesn't use the scheduler

Reply:
I was thinking about the other `townyUniverse.getDataSource().finishTasks();` one but it also looks like it will finish first.